### PR TITLE
WT-9854 Add more cache eviction stats; rename a few existing ones for consistency.

### PR DESCRIPTION
**Pull request by pmacko86**
The PR adds several stats for checking why pages could not be evicted, and it also renames cache_eviction_checkpoint and cache_eviction_hazard to include the word "blocked" for consistency with other such metrics.

I tested a patch build with MongoDB to ensure nothing fails due to the rename.

### CODE REVIEW

File: test/suite/test_stat11.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import wiredtiger
+import wttest
+
+# test_stat11.py
+# Check for the presence of some stats, but does not actually check their values.
```
Comment (etienneptl) [inline test/suite/test_stat11.py:33]: This makes me wonder if we should have a simple python test that walks all of our stats and make sure a cursor stat can be opened on each. What do you think?
If you agree with the idea, a new ticket should be created and every time a new stat is added, the test is updated.

**Review by etienneptl — APPROVED** [review]
Changes LGTM, nice work! I added one comment about a new test that should be done in a separate ticket.

File: test/suite/test_stat11.py (hunk 1 above)
Comment (pmacko86) [reply test/suite/test_stat11.py:33]: It certainly wouldn't hurt! Would it actually make sense to do this as an auto-generated test that would be re-generated every time dist/stat.py is run?

Comment (etienneptl) [reply test/suite/test_stat11.py:33]: Let's discuss with the team if this is worth the effort, I will create a ticket if there's enough interest.

**Pull request merged**